### ISSUE TITLE
update shell completions for borg2

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#7011]
this needs to be done by someone familiar with this, preferably before it gets released.

**Comment by ThomasWaldmann** [comment]
OK, bash was done, thanks to @helmutg!

The other shell completions likely need similar updates.

**Comment by diivi** [comment]
I would like to work on the zsh completions, done with setting things up. Assuming the scope of updation is the same as the bash completions, I can refer the v2 docs and finish this up. Can I work on it?

**Comment by ThomasWaldmann** [comment]
@diivi sure, all open tickets can be worked on.

**Comment by diivi** [comment]
No, I am sorry I forgot to get it unassigned. Before getting this assigned, I thought it was a suggested issue to work on to get started with GSoC, but I later realised it was an idea for the program itself. I shifted my focus to Borgmatic and Vorta and forgot about this, my bad.

**Comment by ThomasWaldmann** [comment]
In case we rename some borg subcommands in b10, the shell completions will also have to get updated accordingly.

**Comment by ThomasWaldmann** [comment]
would be good if some people could help with this after #8292 is closed.

**Comment by ThomasWaldmann** [comment]
I am trying to update the zsh completions...

**Comment by ThomasWaldmann** [comment]
@helmutg I did an update for the bash completions, using PyCharm's Junie AI.

The "repo::archive" completion is still present in the bash code, but borg2 does not do that anymore. It is now separated into --repo R (or BORG_REPO env var) and the archive name is usually a positional argument. Guess this will need someone who is more clever than Junie. :-)

**Comment by ThomasWaldmann** [comment]
The same fix needs to be done for the fish shell completions.

**Comment by ThomasWaldmann** [comment]
Closing due to #9172.
